pkg/lang/ir/v1/system.go

```diff
@@ -46,6 +47,95 @@ func (g generalGraph) compileUbuntuAPT(root llb.State) llb.State {
 	return root
 }
 
+func (g generalGraph) copyAPTSignature(root llb.State, name string, url string) (llb.State, string) {
+	// Download the signature and put it under "/etc/apt/keyrings"
+	var fileName = fmt.Sprintf("%s.asc", name)
+	var filePath = "/etc/apt/keyrings"
+
+	base := llb.Image(builderImage)
+	builder := base.
+		Run(llb.Shlex(fmt.Sprintf("sh -c \"curl %s >> %s\"", url, fileName)),
+			llb.WithCustomName("[internal] downloading apt-source signature in base image")).Root()
+
+	aptSign := root.
+		File(llb.Mkdir(filePath, 0755, llb.WithParents(true)),
+			llb.WithCustomName("[internal] setting target apt-source signature folder")).
+		File(llb.Copy(builder, fileName, filePath+fileName),
+			llb.WithCustomName("[internal] copy signature from builder"))
+
+	return aptSign, filePath + fileName
+}
+
+func (g generalGraph) configRSrc(root llb.State, aptConfig ir.APTConfig, sign string) (llb.State, string) {
+
+	// Created different fields in each line to form up the .sources file
+	// The fields follow the DEB822 format spec
```

Comment:
Follow the go doc standard: https://go.dev/doc/comment

Reply:
Fixed